Add --force option to destructive commands

jastang opened the pull request:
Signed-off-by: Jason Tang <jason@upbound.io>

<!--
Thank you for helping to improve Upbound!

Please read through https://git.io/fj2m9 if this is your first time opening an
Upbound pull request. Find us in https://slack.crossplane.io/messages/upbound if
you need any help contributing.
-->

### Description of your changes
This PR adds a `--force` option to destructive commands, prompting the user by default to confirm the deletion of the entity (repository, robot, robot token, organization). If `--force` is supplied, no prompt is given.

If the user backs out of the delete, we will return an error, which will show the CLI usage by default.

I hope I know what I'm doing! ; )

<!--
Briefly describe what this pull request does. Be sure to direct your reviewers'
attention to anything that needs special consideration.

We love pull requests that resolve an open issue. If yours does, use the below
line to indicate which issue your PR fixes, for example "Fixes #500":
-->

Fixes #232 

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

<!--
Before reviewers can be confident in the correctness of this pull request, it
needs to tested and shown to be correct. Briefly describe the testing that has
already been done or which is planned for this change. Consider pasting snippets
with the commands or configurations you used to test, in case you or a reviewer
needs to repeat the test in future.
-->
Tested commands with and without the `--force` option on dev, except the `org` as I cannot create additional orgs.

hasheddan commented on cmd/up/repository/delete.go:
```diff
@@ -16,17 +16,46 @@ package repository
+
+// AfterApply accepts user input by default to confirm the delete operation.
+func (c *deleteCmd) AfterApply(p pterm.TextPrinter, upCtx *upbound.Context) error {
+	if !c.Force {
+		confirm, err := c.prompter.Prompt("Are you sure you want to delete this repository? [y/n]", false)
+		if err != nil {
+			return err
+		}
+
+		if input.InputYes(confirm) {
+			p.Printfln("Deleting repository %s/%s. This cannot be undone.", upCtx.Account, c.Name)
+		} else {
+			return fmt.Errorf("operation canceled")
+		}
+	}
+	return nil
```
@jastang we typically try to return early to avoid doing extraneous work and the cognitive burden of nested branching (ref: https://github.com/crossplane/crossplane/blob/master/CONTRIBUTING.md#return-early)
```suggestion
	if !c.Force {
		return nil
	}
	confirm, err := c.prompter.Prompt("Are you sure you want to delete this repository? [y/n]", false)
	if err != nil {
		return err
	}

	if input.InputYes(confirm) {
		p.Printfln("Deleting repository %s/%s. This cannot be undone.", upCtx.Account, c.Name)
		return nil
	}
	return fmt.Errorf("operation canceled")
```

hasheddan replied:
Side note: I was trying to decide whether the `operation canceled` scenario should return an error. In some ways it seems like a successful execution of what the user requested. I'm going to look up examples of this in other CLI tools and see how they handle, but did you have any references off the top of your head?

hasheddan replied:
> And is there an easy way to break out of the hook without executing Run()?

If you add a `*kong.Context` you can call `.Exit(0)` in the `AfterApply()`. Similarly, you can call `.PrintUsage()` if that is desirable as well. However, returning the error here might be just fine tbh :)

jastang replied:
Thanks! I'll keep that in my back pocket. Just learning about Kong 🙂

hasheddan commented on cmd/up/repository/delete.go:
```diff
@@ -16,17 +16,46 @@ package repository
+
 	Name string `arg:"" required:"" help:"Name of repository."`
+
+	Force bool `help:"Force deletion of repository." default:"false" short:"f"`
```
This is definitely subjective, but since we use `-f` elsewhere for filepaths _and_ because `--force` is a destructive action, I would probably lean towards making the user type the full command out. What do you think?
```suggestion
	Force bool `help:"Force deletion of repository." default:"false"`
```

jastang replied:
Yeah, I agree with that and am not opinionated (just felt familiar). I can remove :)

hasheddan reviewed (commented):
Thanks so much @jastang! Left a few comments below :)

jastang commented:
@hasheddan addressed your comments and re-tested. I think there's an implicit agreement to tackle `uninstall` separately -- I don't know how "atomic" those operations are, but just preferring not to invite any complexity I might not be able to handle 😅 

thanks for humouring me on this one!

hasheddan reviewed (approved):
Thanks @jastang!
